scene/2d/audio_stream_player_2d.cpp

```diff
@@ -169,116 +64,124 @@ void AudioStreamPlayer2D::_notification(int p_what) {
 	if (p_what == NOTIFICATION_INTERNAL_PHYSICS_PROCESS) {
 		//update anything related to position first, if possible of course
 
-		if (!output_ready.is_set()) {
-			Ref<World2D> world_2d = get_world_2d();
-			ERR_FAIL_COND(world_2d.is_null());
+		if (!stream_playback.is_valid()) {
+			return;
+		}
+		if (setplay.get() >= 0 || (active.is_set() && last_mix_count != AudioServer::get_singleton()->get_mix_count())) {
+			_update_panning();
+			if (setplay.get() >= 0) {
+				active.set();
+				AudioServer::get_singleton()->start_playback_stream(stream_playback, _get_actual_bus(), volume_vector, setplay.get());
+				setplay.set(-1);
+			}
+		}
 
-			int new_output_count = 0;
+		//stop playing if no longer active
+		if (active.is_set() && !AudioServer::get_singleton()->get_is_playback_active(stream_playback)) {
+			active.clear();
+			set_physics_process_internal(false);
+			emit_signal(SNAME("finished"));
+		}
+	}
+}
+StringName AudioStreamPlayer2D::_get_actual_bus() {
+	if (!stream_playback.is_valid()) {
+		return "Master";
```

Comment:
```suggestion
		return SNAME("Master");
```
maybe? Also for other occurrences. `SNAME()` creates and caches the StringName so we always use the same one.